### PR TITLE
Remove whitespace from header field name in http response due to RFC7230:3.2.4 (#6793)

### PR DESCRIPTION
**Pull request by garfieldonly**
We should remove whitespace from header fieldname in http response
#6793

### CODE REVIEW

**Comment by oknet** [comment]
[approve ci]

**Comment by oknet** [comment]
Good catch

**Comment by oknet** [comment]
@garfieldonly Please do clang-format and fix missing comma error.

File: proxy/hdrs/MIME.cc [hunk 1]
```diff
@@ -2570,8 +2570,12 @@ mime_parser_parse(MIMEParser *parser, HdrHeap *heap, MIMEHdrImpl *mh, const char
     // server MUST reject any received request message that contains
     // whitespace between a header field-name and colon with a response code
     // of 400 (Bad Request).
+    //A proxy MUST remove any such whitespace from a response message before
+    // fowarding the message downstream.
     if (is_ws(field_name.back())) {
-      return PARSE_RESULT_ERROR;
+	  if (!remove_ws_from_field_name)
```
Comment (bryancall) [inline proxy/hdrs/MIME.cc:2576]: Please add braces for this if statement.

**Comment by bryancall** [comment]
For reference, this is in the issue, but nice to have here too:
https://tools.ietf.org/html/rfc7230#section-3.2.4
> A proxy MUST remove any such whitespace from a response message before forwarding the message downstream.

File: proxy/hdrs/MIME.cc (hunk 1 above)
Comment (garfieldonly) [reply proxy/hdrs/MIME.cc:2576]: OK!~

**Comment by zwoop** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by oknet** [comment]
[approve ci]

**Comment by oknet** [comment]
[approve ci]

**Comment by garfieldonly** [comment]
@bryancall Please review the changes, thanks!

**Comment by scw00** [comment]
@zwoop any words on this pr?

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

**Pull request merged**